src/components/Filters/Filters.tsx

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import isEqual from 'lodash/isEqual';
+
+import {block} from '../utils/cn';
+import {FieldControl} from './components/FilterControl';
+import {FilterSelect} from './components/FilterSelect';
+import {ClearFiltersButton} from './components/ClearFiltersButton/ClearFiltersButton';
+
+import i18n from './i18n';
+
+import type {FilterSchema} from './schemas/types';
+
+import './Filters.scss';
+
+const b = block('filters');
+
+export type FilterValue<TFilterValue extends any = string[]> = TFilterValue;
+export type FiltersValue<TFiltersValue extends any = Record<string, FilterValue>> = TFiltersValue;
+
+/**
+ * Filters schema to retreive editors/initial values/other schema options.
+ */
+export type FiltersSchema = Record<string, FilterSchema<any, any>>;
+
+interface FiltersComponentProps {
+    className?: string;
+
+    filtersSchema: FiltersSchema;
+
+    showEmptyText?: boolean;
+
+    initialOpenFilterKey?: string;
+
+    /**
+     * Contains serialized filters values with ID as a key.
+     * Filters component uses this object to render selected filters.
+     *
+     * @example {location: ['Moscow'], author: ['Alex']}
+     */
+    initialValues?: FiltersValue;
+    /**
+     * Contains serialized filters values with ID as a key.
+     * Filters component uses this object to render selected filters.
+     *
+     * @example {location: ['Moscow', 'London'], author: ['Max', 'Travis']}
+     */
+    filtersValues: FiltersValue;
+
+    handleChangeFilterValue: (id: string, value: FilterValue) => void;
```

Comment:
Here and below: It is not a good naming for properties. It good practice to use prefix `handle` in method definition inside of component